typpete/src/annotation_resolver.py

```diff
@@ -246,12 +246,17 @@ def add_type_var(self, target, type_var_node, solver, module):
                        fail_message="Upper bound of type variable {}".format(
                            type_var_name))
 
-    def unparse_annotation(self, z3_type):
+    def unparse_annotation(self, z3_type, context_name=None, lineno=0, definition_linenos=None):
         """Unparse the z3_type into a type annotation in PEP 484 syntax
-        
+
         :param z3_type: The z3 type to be unparsed
+        :param context_name: The name of the context this annotation occurs in
+        :param lineno: The line number this annotation is added to
+        :param definition_linenos: The line numbers for type definitions. Used for detecting forward referencing
         :returns str: the unparsed z3_type in PEP 484 syntax
         """
+        if not definition_linenos:
```

Comment:
This looks like you could just make the default argument for ``definition_linenos`` {} instead of None.

Reply:
Using mutable types as default arguments is an anti-pattern in python. Read more about it here https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html

Reply:
Okay, makes sense.